osquery/tables/system/darwin/keychain_utils.cpp

```diff
@@ -121,9 +121,14 @@ void genAlgorithmProperties(X509* cert,
   }
 }
 
-std::string genSHA1ForCertificate(const CFDataRef& raw_cert) {
-  return hashFromBuffer(
-      HASH_TYPE_SHA1, CFDataGetBytePtr(raw_cert), CFDataGetLength(raw_cert));
+std::string genSHA1ForCertificate(X509* cert) {
+  const EVP_MD* fprint_type = EVP_sha1();
+  unsigned char fprint[EVP_MAX_MD_SIZE];
+  unsigned int fprint_size;
+
+  X509_digest(cert, fprint_type, fprint, &fprint_size);
```

Comment:
Perhaps check the result:
```cpp
if (X509_digest(cert, fprint_type, fprint, &fprint_size)) {
  return genKIDProperty(fprint, fprint_size);
}
return "";
```

> X509_digest converts the X509 structure a to DER-encoded form; computes the message digest of it, using as the message digest algorithm; and returns the results in md, with the length of the returned digest in len. 1 is returned on success, or 0 on error.

Not sure if it'd catch/prevent any errors, but worth being clear to only operate on `fprint` if we expect to have it filled int with meaningful data.